File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/CriticalNativeTest.java
package net.kdt.pojavlaunch;

import dalvik.annotation.optimization.CriticalNative;

/**
 * CriticalNativeアノテーションのテストクラス。
 */
public class CriticalNativeTest {
    /**
     * CriticalNativeを使用したネイティブメソッドのテスト。
     */
    @CriticalNative
    public static native void testCriticalNative(int arg0, int arg1);

    /**
     * CriticalNativeテストメソッドを呼び出します。
     */
    public static void invokeTest() {
        testCriticalNative(0, 0);
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/utils/CriticalNativeTest.kt
package com.arata.yukarilauncher.utils

import dalvik.annotation.optimization.CriticalNative

/**
 * CriticalNativeアノテーションを使用したJNI呼び出しのテストを行うオブジェクト。
 * ARTランタイムでのCriticalNative最適化が正しく機能するかを検証します。
 */
object CriticalNativeTest {
    /**
     * CriticalNativeアノテーションが付与されたテスト用ネイティブメソッド。
     * @param arg0 テスト引数1
     * @param arg1 テスト引数2
     */
    @CriticalNative
    @JvmStatic external fun testCriticalNative(arg0: Int, arg1: Int)

    /**
     * CriticalNativeテストを実行します。
     */
    @JvmStatic
    fun invokeTest() {
        testCriticalNative(0, 0)
    }
}